storage/kvstore.go

```diff
@@ -489,3 +489,29 @@ func (s *store) delete(key []byte) {
 	}
 	s.currentRev.sub += 1
 }
+
+// appendMarkTombstone appends tombstone mark to the backend key.
```

Comment:
My interpretation is that tombstone is a type, but that this is a mark indicating that this key is of type tombstone. So a revision is a normal kv (of type KV) if it has no mark, but a tombstone kv (of type tombstone KV) if it has a mark. Does that make sense?